### ISSUE TITLE
No streaming

### ISSUE THREAD

**Issue by mlcarson** [fHDHR/fHDHR_USTVGO#118]
USTVGO works in a web browser.   19 channels missing without VPN.  All channels present and working with VPN.
Streaming simply doesn't happen with this app.  The equivalent fHDHR_Locast and fHDHR_Pluto apps work without issue.  Playback method is direct -- standard defaults.
I get:
fHDHR - INFO - Wrote ustvgo EPG cache. 33880 Programs for 88 Channels
fHDHR - INFO - Tuner #0 Acquired.
seleniumwire.proxy.backend - INFO - Created proxy listening on 127.0.0.1:36219
About 90 seconds later:
Traceback (most recent call last):
  File "/home/mlcarson/.local/lib/python3.7/site-packages/gevent/pywsgi.py", line 999, in handle_one_response
    self.run_application()
  File "/home/mlcarson/.local/lib/python3.7/site-packages/gevent/pywsgi.py", line 945, in run_application
    self.result = self.application(self.environ, self.start_response)
  File "/home/mlcarson/.local/lib/python3.7/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/home/mlcarson/.local/lib/python3.7/site-packages/flask/app.py", line 1867, in handle_exception 
reraise(exc_type, exc_value, tb)
  File "/home/mlcarson/.local/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
  raise value
  File "/home/mlcarson/.local/lib/python3.7/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/mlcarson/.local/lib/python3.7/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/mlcarson/.local/lib/python3.7/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/mlcarson/.local/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/mlcarson/.local/lib/python3.7/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/mlcarson/.local/lib/python3.7/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/mlcarson/Downloads/fHDHR_USTVGO-main/fHDHR_web/api/tuners.py", line 20, in __call__
    return self.get(*args)
  File "/home/mlcarson/Downloads/fHDHR_USTVGO-main/fHDHR_web/api/tuners.py", line 90, in get
    stream_args = self.fhdhr.device.tuners.get_stream_info(stream_args)
  File "/home/mlcarson/Downloads/fHDHR_USTVGO-main/fHDHR/device/tuners/__init__.py", line 95, in get_stream_info
    stream_info = self.channels.get_channel_stream(stream_args)
  File "/home/mlcarson/Downloads/fHDHR_USTVGO-main/fHDHR/device/channels/__init__.py", line 115, in get_channel_stream
    return self.origin.get_channel_stream(self.get_channel_dict("number", stream_args["channel"]), stream_args)
  File "/home/mlcarson/Downloads/fHDHR_USTVGO-main/fHDHR/originwrapper/__init__.py", line 40, in get_channel_stream
    return self.channels.get_channel_stream(chandict, stream_args)
  File "/home/mlcarson/Downloads/fHDHR_USTVGO-main/origin/origin_channels.py", line 56, in get_channel_stream
    streamurl = self.get_ustvgo_stream(chandict)
  File "/home/mlcarson/Downloads/fHDHR_USTVGO-main/origin/origin_channels.py", line 63, in get_ustvgo_stream
    driver = self.get_firefox_driver()
  File "/home/mlcarson/Downloads/fHDHR_USTVGO-main/origin/origin_channels.py", line 121, in get_firefox_driver
    driver = webdriver.Firefox(seleniumwire_options=set_seleniumwire_options, options=ff_options, firefox_profile=firefox_profile)
  File "/home/mlcarson/.local/lib/python3.7/site-packages/seleniumwire/webdriver/browser.py", line 48, in __init__
    super().__init__(*args, **kwargs)
  File "/home/mlcarson/.local/lib/python3.7/site-packages/selenium/webdriver/firefox/webdriver.py", line 174, in __init__
    keep_alive=True)
  File "/home/mlcarson/.local/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 157, in __init__
    self.start_session(capabilities, browser_profile)
  File "/home/mlcarson/.local/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 252, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
  File "/home/mlcarson/.local/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/home/mlcarson/.local/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: newSession

2021-01-21T22:05:54Z {'REMOTE_ADDR': '192.168.1.101', 'REMOTE_PORT': '56978', 'HTTP_HOST': '192.168.1.101:5004', (hidden keys: 22)} failed with WebDriverException

fHDHR - INFO - 192.168.1.101 - - [2021-01-21 14:05:54] "GET /api/tuners?method=direct&channel=1 HTTP/1.1" 500 161 72.436252
fHDHR - INFO - Tuner #1 Acquired.

**Comment by deathbybandaid** [comment]
This variant works for some people and not others, and for no good reason.

The stream has to be scraped from the webpage, as there's no API for this.

**Comment by deathbybandaid** [comment]
I have it working by passing through PIA VPN to New York, but it wouldn't wrok from my IP.

**Comment by deathbybandaid** [comment]
I just retested several channels and it works

**Comment by mlcarson** [comment]
I know USTVGO is doing whitelisting/blacklisting and it seems to be on a per channel basis but the VPN I tried earlier had all of the channels working but yet the APP still failed.   The VPN endpoint was in Los Angeles rather than NY but most channels work for me even without a VPN.   

The only other question mark that I have in my own setup is the Python version since I'm using Debian.  I'll double check that and maybe try it within Docker just to verify the same result.   I know I installed an older version of Geckodriver because of that.   And I think that's a dependency that's not listed or maybe I just saw a different error earlier that led me to that conclusion.
